Amz.py

```diff
@@ -42,6 +41,39 @@ def pause_small():
     time.sleep(1)
 
 
+def give_away_type(driver):
+    try:
+        driver.find_element(By.XPATH, '//div[@class="amazon-video"]').click()
+        pause_small()
+        submit = WebDriverWait(driver, 20).until(
+            EC.element_to_be_clickable(
+                (By.XPATH, '//button[@class="a-button a-button-primary amazon-video-continue-button"]'))
+        )
+        submit.click()
+    except:
+        try:
+            js = 'document.evaluate("//div[@class=\'youtube-video\']/a", document.body, null, 9, null). singleNodeValue.click();'
+            driver.execute_script(js)
+            pause_small()
+            submit = WebDriverWait(driver, 20).until(
+                EC.element_to_be_clickable(
+                    (By.XPATH, '//button[@class="a-button a-button-primary youtube-continue-button"]'))
+            )
+            submit.click()
+        except:
+            try:
+                driver.find_element(By.XPATH, '//*[@class="a-text-center box-click-area"]').click()
+            except:
+                print('Could not determine giveaway type')
+                x = 1
+                return x
```

Comment:
This seems weird. Are you only returning a value if you can't determine giveaway type? Why not return a Boolean, and use False if can't determine giveaway type?

Reply:
Fixed with the new Giveaway Type logic.
See if that makes more sense